contracts/generator_controller/src/contract.rs

```diff
@@ -405,6 +423,29 @@ fn tune_pools(deps: DepsMut, env: Env) -> ExecuteResult {
         config.pools_limit,
     )?;
 
+    // Set allocation points for the main pool
+    if let Some(main_pool) = config.main_pool {
+        // Main pool may appear in the pool list thus we need to eliminate its contribution in the total VP
+        tune_info
```

Comment:
The main pool should not affect the number of fields. We will always use only **pools_limit - 1** pools

Reply:
It can affect number of pools_limit exactly by 1 according to the task.

Reply:
I got the meaning. Thank you. Fixed